assets/js/utils/rtcUtils.ts

```diff
@@ -184,3 +159,124 @@ export const leaveWebrtc = (webrtc: MembraneWebRTC, clientName: string, webrtcCh
   removeStream(clientName);
   webrtc.leave();
 };
+
+export const shareScreen = async (
+  webrtc: MembraneWebRTC,
+  clientName: string,
+  playerCallback: (SourceType: SourceType) => void
+) => {
+  mergedScreenRef.refreshId && removeMergedStream();
+  const mergedStream = await getMergedTracks(clientName);
+
+  mergedStream.getTracks().forEach((track) => {
+    addOrReplaceTrack(clientName, track, playerCallback);
+  });
+
+  const newTrack = findTrackByType(clientName, "video");
+
+  if (!webrtc || !newTrack) return;
+  if (sourceIds["video"]) webrtc.replaceTrack(sourceIds["video"], newTrack);
+  else sourceIds["video"] = webrtc.addTrack(newTrack, presenterStreams[clientName]);
+};
+
+const getMergedTracks = async (clientName: string) => {
+  const screenStream: MediaStream = await navigator.mediaDevices.getDisplayMedia(VIDEO_CONSTRAINTS);
+  const cameraStream = new MediaStream(presenterStreams[clientName].getVideoTracks()).clone();
+
+  mergedScreenRef.tracks.push(...screenStream.getTracks(), ...cameraStream.getTracks());
+
+  const camera = await attachToDOM("justCamera", cameraStream);
+  const screen = await attachToDOM("justScreenShare", screenStream);
+
+  let canvasElement = document.createElement("canvas");
+  let canvasCtx = canvasElement.getContext("2d");
+
+  canvasCtx
+    ? await makeComposite(canvasElement, canvasCtx, camera, screen)
+    : console.error("CanvasCtx is null", canvasCtx);
+
+  return canvasElement.captureStream(FRAME_RATE);
+};
+
+const attachToDOM = async (id: string, stream: MediaStream) => {
+  let videoElem = document.createElement("video");
+  videoElem.id = id;
+  videoElem.width = 1280;
+  videoElem.height = 720;
```

Comment:
I think this `videoElem` should have the same dimensions as in this in `VIDEO_CONSTRAINTS`